src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -748,6 +747,33 @@ public ChildSyntaxList ChildNodesAndTokens()
             return new ChildSyntaxList(this);
         }
 
+        internal ChildSyntaxList.Segment SiblingsAfterSelf()
+        {
+            int initialIndex = 0;
+            int terminalIndex = 0;
+            if (_parent != null)
+            {
+                initialIndex = _parent.GetIndexOfChild(this);
+                Debug.Assert(initialIndex >= 0, "Node couldn't be found in its parent.");
+                initialIndex &= int.MaxValue; // Doesn't affect non-negative numbers.
+                                              // Turns the not-found index (-1) into a virtually unreachable number, causing the segment to be empty.
+
+                terminalIndex = ChildSyntaxList.CountNodes(_parent.Green);
+            }
+            return new ChildSyntaxList.Segment(_parent!, initialIndex, terminalIndex);
+        }
+
+        internal ChildSyntaxList.Reversed SiblingsBeforeSelf()
```

Comment:
needs to doc that this seems to eb iterating in the backward direction.  in other words you get the immediatley prior sibling, then the one before that, and so on.  Otehrwise, it seems like you could be getting the siblings (in order) *up to* this node.